File: solutions/day-01-concatenation-of-array/Solution.py
from typing import List


class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * (2 * n)

        for idx in range(n):
            result[idx] = nums[idx]
            result[idx + n] = nums[idx]

        return result

File: solutions/day-01-concatenation-of-array/Solution.cpp
#include <vector>

class Solution {
public:
    std::vector<int> getConcatenation(std::vector<int>& nums) {
        int n = static_cast<int>(nums.size());
        std::vector<int> result(2 * n);

        for (int i = 0; i < n; ++i) {
            result[i] = nums[i];
            result[i + n] = nums[i];
        }

        return result;
    }
};
